catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/FindChildrenStreamEndPlugin.java

```diff
@@ -0,0 +1,176 @@
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+
+/**
+ * Find the children of the parent metacard and process the children in batches. The children will be
+ * processed in the order in which they were originally created. If an exception occurs during when a
+ * batch is being retrieved from the CatalogFramework or from a batch handler, then the {@link Handler#end(Context, Metacard)}
+ * will not be called.
+ */
+public class FindChildrenStreamEndPlugin implements StreamEndPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FindChildrenStreamEndPlugin.class);
+
+    private static final int BATCH_SIZE = 50;
+
+    private final Factory factory;
+
+    private final FilterBuilder filterBuilder;
+
+    public FindChildrenStreamEndPlugin(FilterBuilder filterBuilder, Factory factory) {
+        this.filterBuilder = filterBuilder;
+        this.factory = factory;
+    }
+
+    public Factory getFactory() {
+        return factory;
+    }
+
+    @Override
+    public void streamEnded(Context context) {
+
+        context.modifyParentOrChild(new Function<AtomicBoolean, Void>() {
+            @Override
+            public Void apply(AtomicBoolean isParentDirty) {
+
+                if (isParentDirty.get()) {
+                    context.getParentMetacard()
+                            .ifPresent(parentMetacard -> handleParentMetacard(context,
+                                    parentMetacard));
+
+                    isParentDirty.set(false);
+                }
+
+                return null;
+            }
+        });
+
+    }
+
+    @Override
+    public void accept(Visitor visitor) {
+        visitor.visit(this);
+    }
+
+    private void handleParentMetacard(Context context, Metacard parentMetacard) {
+
+        Handler handler = factory.build();
+
+        try {
+
+            Filter filter = filterBuilder.attribute(Associations.DERIVED)
+                    .is()
+                    .equalTo()
+                    .text(parentMetacard.getId());
+
+            int startIndex = 1;
+            int expectedReturnCount = 0;
+
+            do {
+                QueryRequest queryRequest = new QueryRequestImpl(new QueryImpl(filter,
+                        startIndex,
+                        BATCH_SIZE,
+                        new SortByImpl(Core.METACARD_CREATED, SortOrder.ASCENDING),
+                        true,
+                        0));
+
+                QueryResponse queryResponse = context.getUdpStreamProcessor()
+                        .getCatalogFramework()
+                        .query(queryRequest);
+
+                if (startIndex == 1) {
+                    expectedReturnCount = Math.toIntExact(queryResponse.getHits());
```

Comment:
Changed it. It's kind of weird, QueryImpl requires an int for the start index, but queryResponse.getHits() returns a long. Seems like those should use the same type.